### DISCUSSION TITLE
[Proposal] Get rid of external_fts pipeline

### DISCUSSION THREAD

**Discussion by avamingli** [apache/cloudberry#627] [Proposal] [answered]
### Proposal Status

Under Discussion

### Abstract

Remove external_fts CI.

### Motivation

The external_fts CI is suggested by me to check a explain which have Entry locus.
It has the same cases with other CI except the `gp_segment_configuration` is stored in ETCD, only this one, other catalogs are in postgres.
Nowadays the https://github.com/cloudberrydb/cloudberrydb/pull/294 cases is changed to use `pg_class`  instead of `gp_segment_configuration`.
We have no plan to check Entry locus using `gp_segment_configuration`.
And, the extra calls with ETCD makes our CI slower, 
According to my observation, external_fts pipeine takes about extra 30min than normal ones(planner, ORCA). An example

<img width="1297" alt="image" src="https://github.com/user-attachments/assets/dcea999d-756e-422f-b82f-e537a31b6f1d">

exteranl_fts is definitely a short board of the barrel. It takes more time and cost.

### Implementation

Remove CI.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by edespino** [comment]
@avamingli, are you referring to getting ride of the CI [build_external_fts.yml](https://github.com/cloudberrydb/cloudberrydb/blob/main/.github/workflows/build_external_fts.yml) (GitHub Action workflow) pipeline?

**Reply by avamingli** [reply]
Yes. Added by  commit https://github.com/cloudberrydb/cloudberrydb/commit/1dfe98e4695121d991bbf2e7152320937afdb49f

**Reply by edespino** [reply]
Will there be a pipeline will be run on some interval to validate the external fts configuration?

**Reply by avamingli** [reply]
> on some interval

Do you mean release pipeline before each release version?

**Reply by edespino** [reply]
Yes, that's correct. As it stands, I am not aware of any release tests being performed on the open-source side specifically for external FTS. This presents a risk because contributors from the community will have no way of knowing if they impact important features like external FTS in the absence of a dedicated validation process. I would suggest considering some form of periodic or pre-release testing to ensure that external FTS functionality remains intact, especially since it touches critical components like gp_segment_configuration and involves ETCD, which may not be covered by standard CI pipelines.

**Reply by avamingli** [reply]
Yes, we have external_fts in internal release pipeline of HashData Lightning. 
We could do the same, move it to Github Release pipeline to avoid overhead during dev.

**Reply by edespino** [reply, marked as answer]
@avamingli That is a great idea proposing we incorporate that to a Open Source release pipeline or possibly weekly run pipeline. I am in agreement the overhead to run it for each commit is too much.
